src/main/java/jenkins/scm/api/SCMEvent.java

```diff
@@ -541,17 +540,16 @@ public void run() {
                         event.getClass(), event.getTimestamp(), oldName)
                 );
                 for (final SCMEventListener l : ExtensionList.lookup(SCMEventListener.class)) {
-                    SecurityContext context = ACL.impersonate(ACL.SYSTEM);
-                    try {
-                        fire(l, event);
-                    } catch (LinkageError e) {
-                        log(l, e);
-                    } catch (Error e) {
-                        throw e;
-                    } catch (Throwable e) {
-                        log(l, e);
-                    } finally {
-                        SecurityContextHolder.setContext(context);
+                    try (ACLContext ctx = ACL.as2(ACL.SYSTEM2)) {
+                        try {
+                            fire(l, event);
```

Comment:
Baseline is new enough now to use https://javadoc.jenkins.io/jenkins/util/Listeners.html#notify(java.lang.Class,boolean,java.util.function.Consumer) actually.